var/spack/repos/builtin/packages/opencv/package.py

```diff
@@ -32,62 +32,52 @@ class Opencv(CMakePackage, CudaPackage):
-
-    # OpenCV modules
-    variant('calib3d', default=False, description='calib3d module')
-    variant('core', default=True, description='Include opencv_core module into the OpenCV build')
-    variant('cudacodec', default=False, description='Enable video encoding/decoding with CUDA')
-    variant('dnn', default=False, description='Build DNN support')
-    variant('features2d', default=False, description='features2d module')
-    variant('flann', default=False, description='flann module')
-    variant('highgui', default=False, description='Include opencv_highgui module into the OpenCV build')
-    variant('imgcodecs', default=False, description='Include opencv_imgcodecs module into the OpenCV build')
-    variant('imgproc', default=False, description='Include opencv_imgproc module into the OpenCV build')
-    variant('java', default=False,
-            description='Activates support for Java')
-    variant('ml', default=False, description='Build ML support')
-    variant('python', default=False,
-            description='Enables the build of Python extensions')
-    variant('stitching', default=False, description='stitching module')
-    variant('superres', default=False, description='superres module')
-    variant('ts', default=False, description='Include opencv_ts module into the OpenCV build')
-    variant('video', default=False, description='video module')
-    variant('videostab', default=False, description='videostab module')
-    variant('videoio', default=False, description='videoio module')
-
-    # Optional 3rd party components
-    variant('eigen', default=False, description='Activates support for eigen')
-    variant('ipp', default=False, description='Activates support for IPP')
-    variant('ipp_iw', default=False, description='Build IPP IW from source')
-    variant('jasper', default=False, description='Activates support for JasPer')
-    variant('jpeg', default=False, description='Include JPEG support')
-    variant('opencl', default=False, description='Include OpenCL Runtime support')
-    variant('opencl_svm', default=False, description='Include OpenCL Shared Virtual Memory support')
-    variant('openclamdfft', default=False, description='Include OpenCL AMD OpenCL FFT library support')
-    variant('openclamdblas', default=False, description='Include OpenCL AMD OpenCL BLAS library support')
-    variant('openmp', default=False, description='Activates support for OpenMP threads')
-    variant('pthreads_pf', default=False, description='Use pthreads-based parallel_for')
-    variant('png', default=False, description='Include PNG support')
-    variant('qt', default=False, description='Activates support for QT')
-    variant('gtk', default=False, description='Activates support for GTK')
-    variant('tiff', default=False, description='Include TIFF support')
-    variant('vtk', default=False, description='Activates support for VTK')
-    variant('zlib', default=False, description='Build zlib from source')
```

Comment:
Is `zlib` always required now

Reply:
Correct, I couldn't find a way to build OpenCV without zlib support. Even if I disable all of the variants, it still links to the system zlib.